### DISCUSSION TITLE
Problem getting sonoff basic r2 connect to wifi

### DISCUSSION THREAD

**Discussion by psu** [RavenSystem/esp-homekit-devices#1527] [Q&A] [answered]
Hi,
I have tried to install a sonoff basic r2 with no luck. JSON below. I have installed several Sonoff Mini (with a different JSON) without any trouble. Am I missing something?

`{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}],"s":0}]}`

**Comment by huddx01** [comment]
What do you mean with, "Problem getting it to wifi"?
Your basic is already connected to wifi, or not?

What exactly do you want to realize with this json?

**Reply by psu** [reply]
Sorry, I will be a little more elaborate: I have flashed it with the latest version of HAA (not OTA) and I am accessing it in setup mode using the device own access point wifi (HAA-xxxx).

I select my wifi SSID, enter a password, paste the JSON and tap Save. Get the usual “Ok” message and wait for about five minutes for the setup to take effect. 

Now I expect it to exit setup mode and connect to the selected SSID. Then I usually add it as a HomeKit accessory. 

--- 
The JSON is supposed to tell HomeKit that it is a switch type device with on and off state. Is this not correct?

**Comment by huddx01** [comment]
I think it does not accept your json config in AP Mode...

I think you forgot to add the  "v": 0 Parameter in Off-State...?

Try this...

```
{
  "c": {
    "l": 13,
    "b": [
      {
        "g": 0,
        "t": 5
      }
    ]
  },
  "a": [
    {
      "0": {
        "r": [
          {
            "g": 12,
            "v": 0
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 12,
            "v": 1
          }
        ]
      },
      "b": [
        {
          "g": 0
        }
      ],
      "s": 0
    }
  ]
}
```

**Comment by huddx01** [comment]
Or fist use the json from the wiki... then bring it to your wifi... and the modify the json as you need...

[https://github.com/RavenSystem/esp-homekit-devices/wiki/Switches#example](https://github.com/RavenSystem/esp-homekit-devices/wiki/Switches#example)

**Reply by psu** [reply]
I can try with the example JSON, maybe it gives me luck. 
The v:0 is omitted in my code since it is a default value, but could also try with it set explicitly ofc

**Reply by huddx01** [reply]
Hmmm. okay... then try the example first. For me, it works.

**Comment by psu** [comment, marked as answer]
After some more fiddling my conclusion is that the device is broken somehow. After flashing it again I can't connect to it at all. Thanks for the help anyway!
